### PR TITLE
2.x: Update OCI integration documentation to reflect use of new OCI SDK extension

### PR DESCRIPTION
**Pull request by barchetta**
See #4124

### CODE REVIEW

File: docs/mp/oci/01_oci.adoc [hunk 1]
```diff
@@ -16,63 +16,76 @@
...
+The extension also handles authenticating with OCI by automatically picking up OCI
+credentials from your environment.
 
-===  Using Helidon MP Properties Configuration
+== Configuring the Helidon OCI SDK Extension
 
-The first option to configure connection to OCI is to directly specify properties in `microprofile-config.properties` file:
+When you inject an OCI SDK Client object, the Helidon OCI SDK extension
+configures and constructs the object for you. The configuration primarily
+consists of initializing an OCI Authentication Details Provider. By default
```
Comment (ljnelson) [inline docs/mp/oci/01_oci.adoc:60]: Nit: should be `AuthenticationDetailsProvider`; it's a classname

File: docs/mp/oci/03_vault.adoc [hunk 2]
```diff
@@ -16,267 +16,89 @@
...
+    <artifactId>oci-java-sdk-keymanagement</artifactId>
+</dependency>
+<dependency>
+    <groupId>com.oracle.oci.sdk</groupId>
+    <artifactId>oci-java-sdk-secrets</artifactId>
+</dependency>
 ----
 
-==== Creating a Signature
+== Injecting an Vault client
```
Comment (ljnelson) [inline docs/mp/oci/03_vault.adoc:63]: Typo: should be "a"

File: docs/mp/oci/04_atp.adoc [hunk 3]
```diff
@@ -15,135 +15,70 @@
...
-The Helidon MP OCI Autonomous Transaction Processing integration provides easy access to ATP database in Oracle cloud.
-
-== Experimental
+:feature-name: OCI atp
+:oci-atp-javadoc: https://docs.oracle.com/en-us/iaas/tools/java/latest/com/oracle/bmc/database/package-summary.html
+:oci-atp-url: https://docs.oracle.com/en-us/iaas/Content/Database/home.htm
 
-WARNING: Helidon integration with Oracle Cloud Infrastructure is still experimental and not intended for production use. APIs and features have not yet been fully tested and are subject to change.
+You can use Helidon's <<mp/oci/01_oci.adoc,OCI SDK Extension>>
+to access OCI Services. This document describes how to use it to access
```
Comment (ljnelson) [inline docs/mp/oci/04_atp.adoc:30]: Nit: access **the**

File: docs/se/oci/03_vault.adoc [hunk 4]
```diff
@@ -24,12 +24,24 @@
 :keywords: oci, vault
 :common-deps-page-prefix-inc: ../../shared/dependencies/common_shared.adoc
 :feature-name: OCI Vault
+:oci-vault-javadoc: https://docs.oracle.com/en-us/iaas/tools/java/latest/com/oracle/bmc/vault/package-summary.html
+:oci-vault-url: https://docs.oracle.com/en-us/iaas/Content/KeyManagement/home.htm
+:helidon-tag: https://github.com/oracle/helidon/tree/{helidon-version}
 
 The Helidon SE OCI Vault integration provides a reactive API Oracle Cloud Vault features.
```
Comment (ljnelson) [inline docs/se/oci/03_vault.adoc:31]: I know you didn't edit this line but it doesn't parse.

**Review by ljnelson — APPROVED** [review]
A couple of minor points to fix or not; otherwise looks fine

File: docs/mp/oci/01_oci.adoc (hunk 1 above)
Comment (barchetta) [reply docs/mp/oci/01_oci.adoc]: Fixed

File: docs/mp/oci/04_atp.adoc (hunk 3 above)
Comment (barchetta) [reply docs/mp/oci/04_atp.adoc]: Fixed

File: docs/se/oci/03_vault.adoc (hunk 4 above)
Comment (barchetta) [reply docs/se/oci/03_vault.adoc]: Fixed

File: docs/mp/oci/01_oci.adoc [hunk 5]
```diff
@@ -57,7 +57,7 @@ credentials from your environment.
 
 When you inject an OCI SDK Client object, the Helidon OCI SDK extension
 configures and constructs the object for you. The configuration primarily
-consists of initializing an OCI Authentication Details Provider. By default
+consists of initializing an OCI `AuthenticationDetailsProvider`. By default
 the extension will examine your environment and select the best Authentication
```
Comment (ljnelson) [inline docs/mp/oci/01_oci.adoc:61]: (Technically you should change this reference too; up to you.)

**Review by ljnelson — APPROVED** [review]
One tiny nit; fix or not; otherwise looks good

**Review by ljnelson — APPROVED** [review]

File: docs/mp/oci/03_vault.adoc (hunk 2 above)
Comment (barchetta) [reply docs/mp/oci/03_vault.adoc]: Fixed

File: docs/mp/oci/01_oci.adoc (hunk 5 above)
Comment (barchetta) [reply docs/mp/oci/01_oci.adoc]: Fixed

**Pull request merged**